src/App.tsx

```diff
@@ -14,11 +16,86 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGood: string[],
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  clearSelectedGood = () => {
+    this.setState({ selectedGood: [] });
+  };
+
+  changeSelectedGood = (good: string) => {
+    this.setState((prevState) => ({ selectedGood: [...prevState.selectedGood, good] }));
+  };
+
+  createTitle = () => {
+    const { selectedGood } = this.state;
+
+    switch (selectedGood.length) {
+      case 0:
+        return 'No goods selected';
+
+      case 1:
+        return `${selectedGood[0]} is selected`;
+
+      case 2:
+        return `${selectedGood.join(' and ')} are selected`;
+
+      default:
+        return `${selectedGood.slice(0, -2).join(', ')}, ${selectedGood.slice(-2).join(' and ')} are selected`;
+    }
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="app">
+        <section className="products">
+          <h1 className="products__title">
+            <button
+              type="button"
+              className={classNames('products__close', { active__close: selectedGood.length === 0 })}
+              disabled={selectedGood.length === 0}
+              onClick={this.clearSelectedGood}
+            >
+              X
+            </button>
+            <>
+              {this.createTitle()}
```

Comment:
you can call this function before return and assign its return value to a variable. and give it appropriate name, so that it is clear what it is